### PR TITLE
chore: Add jextract candidate

### LINKED ISSUE #663: Feature: Add support for `jextract`
- [x] I have read the [CONTRIBUTING guidelines](CONTRIBUTING.md)
- [ ] I've had a conversation on the [#cli-development](https://sdkman.slack.com/app_redirect?channel=cli-development) Slack channel about this request.

**Feature request**

The `jextract` tool used to be part of the JDK in older builds of Project Panama but [has been externalized](https://github.com/openjdk/jextract) to decouple the release cycle. It helps generating Java bindings from C header files using the FFM API (currently still in preview).

You might want to have multiple versions of jextract installed at the same time, each of which compatible with a different JDK. SDKMAN! would simplify managing those installations.

### PR DESCRIPTION
**Pull request by helpermethod**
Fixes #663.

### CODE REVIEW

**Review by marc0der — CHANGES_REQUESTED** [review]
Thanks @helpermethod! We first need to add support for the tgz files in the sdkman-hooks project. Marking this as "Request changes" until we have this in place.

**Review by marc0der — APPROVED** [review]

File: src/main/scala/io/sdkman/changelogs/JExtractMigrations.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    order = "001",
+    id = "001_add_jextract_candidate",
+    author = "helpermethod"
+  )
+  def migration001(implicit db: MongoDatabase) =
+    Candidate(
+      candidate = "jextract",
+      name = "Jextract",
+      description =
+        "jextract is a tool which mechanically generates Java bindings from a native library headers.",
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/JExtractMigrations.scala:19]: I think the official name is `Jextract`. Mind if we change that?

Comment (marc0der) [reply src/main/scala/io/sdkman/changelogs/JExtractMigrations.scala:19]: Never mind :laughing:

Comment (helpermethod) [reply src/main/scala/io/sdkman/changelogs/JExtractMigrations.scala:19]: I've just copied the description from the GH page 😄

**Pull request merged**